docs/lang/articles/contribution/dev_install.md

```diff
@@ -4,318 +4,467 @@ sidebar_position: 2
+
+# If you are building on Apple M1, use -DLLVM_TARGETS_TO_BUILD="AArch64".
+
+# If you are building on NVIDIA Jetson TX2, use -DLLVM_TARGETS_TO_BUILD="ARM;NVPTX"
+
+# If you are building for a PyPI release, add -DLLVM_ENABLE_Z3_SOLVER=OFF to reduce the library dependency.
+
+make -j 8
+
+sudo make install
 
-If you wish to build taichi with Vulkan. You will need to install the Vulkan SDK. Please visit [this website](https://vulkan.lunarg.com/sdk/home) and follow the instructions for your OS.
-- If you are working on Windows, please set the environment variable `VULKAN_SDK` to `C:/VulkanSDK/${YOUR_VULKAN_VERSION}`. (For example, when using Vulkan 1.2.189.0, set `VULKAN_SDK` to `C:/VulkanSDK/1.2.189.0`).
-- On Linux, also make sure the environment variable `VULKAN_SDK` `PATH` `LD_LIBRARY_PATH` and `VK_LAYER_PATH` are updated. On Ubuntu, the downloaded SDK provides a `setup-env.sh` that can be sourced.
-- Make sure you have a Vulkan driver from a GPU vendor installed. On Ubuntu, you
-  can verify there is a JSON file in one of these two locations: `/etc/vulkan/icd.d/` or `/usr/share/vulkan/icd.d`.
-- You can verify the installation of the Vulkan SDK by running `vkvia`, `vulkaninfo`, and/or `vkcube`.
+# Check your LLVM installation
 
-After Vulkan is successfully installed. You can build Taichi with Vulkan by adding an environment variable `TAICHI_CMAKE_ARGS` with the value `-DTI_WITH_VULKAN:BOOL=ON`.
+llvm-config --version  # You should get 10.0.0
+```
+
+</TabItem>
+
+<TabItem value="windows">
+
+```shell
+# For Windows
+
+# LLVM 10.0.0 + MSVC 2019
+
+cmake .. -G "Visual Studio 16 2019" -A x64 -DLLVM_ENABLE_RTTI:BOOL=ON -DBUILD_SHARED_LIBS:BOOL=OFF   -DCMAKE_BUILD_TYPE=Release -DLLVM_TARGETS_TO_BUILD="X86;NVPTX" -DLLVM_ENABLE_ASSERTIONS=ON -Thost=x64   -DLLVM_BUILD_TESTS:BOOL=OFF -DCMAKE_INSTALL_PREFIX=installed
+```
+
+1. Use Visual Studio 2017+ to build **LLVM.sln**.
+2. Ensure that you use the **Release** configuration. After building the `INSTALL` project (under folde **CMakePredefinedTargets** in the Solution Explorer window). 
+3. If you use MSVC 2019, ensure that you use **C++17** for the `INSTALL` project.
+4. When the build completes, find your LLVM binaries and header files in **build/installed**. Add `build/installed/bin` to `PATH`. Later, when you build Taichi, please use `cmake -DLLVM_DIR=&lt;PATH_TO_BUILD&gt;/build/installed/lib/cmake/llvm`.
+
+</TabItem>
+
+</Tabs>
 
-## Building Taichi from source
+</details>
 
-1. Clone the Taichi repo **recursively**, and build:
+## Build Taichi from source
 
-  ```bash
+<Tabs
+  defaultValue="linux"
+  values={[
+    {label: 'Linux & macOS', value: 'linux'},
+    {label: 'Windows', value: 'windows'},
+  ]}>
+
+<TabItem value="linux">
+
+1. Clone the Taichi repo *recursively* and build[^1]:
+
+  ```shell
   git clone --recursive https://github.com/taichi-dev/taichi
+
   cd taichi
+
```

Comment:
Better to create a virtual env for Taichi?

Reply:
@skyzh Do you mean we need to add steps for creating a virtual environment. IMO, it is a given so I omitted it. Let me know how you think. 

Cheers
Herman

Reply:
Yes, I believe developers don't only develop Taichi in their system. It's better to use a separate virtualenv for development.

Reply:
Gotcha. Thanks for the clarification. We'll add such a section in 22Q1.